internal/database/sql/sql_test.go

```diff
@@ -0,0 +1,16 @@
+package sql
+
+import (
+	"testing"
+	"tracker/internal/database"
+)
+
+func TestImplements(t *testing.T) {
+	var i interface{} = &UserDatabase{}
+
+	if _, ok := i.(database.UserDatabase); !ok {
+		t.Errorf("UserDatabase doesn't implement database.UserDatabase")
+	}
```

Comment:
We can replace this whole thing with a compile-time check:
```go
var _ database.UserDatabase = (*UserDatabase)(nil)
```

Reply:
I've seen this in use, however in my opinion it directly contradicts 

>There is no explicit declaration of intent, no "implements" keyword.
>
> Implicit interfaces decouple the definition of an interface from its implementation
- https://tour.golang.org/methods/10.

The compile-time check adds this explicit declaration of intent. Placing it in tests allows to check can the type potentially be used as intended, rather than tying it to the interface. Furthermore, when a consumer is using the type, they will get that compile-time check anyways.

But this is just my opinion, if you prefer for me to change it, I'll be happy to oblige.